addons/web/static/src/scss/modal.scss

```diff
@@ -1,5 +1,13 @@
 .modal.o_technical_modal {
+    @include media-breakpoint-down(sm) {
+        .modal-dialog {
+            margin: 0 !important;
 
+            .modal-content {
+                border: 0px solid rgba(0, 0, 0, 0.2);
+            }
+        }
+    }
```

Comment:
@koz-odoo Nope, you should try utilising [fullscreen](https://github.com/odoo/odoo/blob/b51129bd304e74a5eef94fcabca915aedf56e5ad/addons/web/static/src/js/core/dialog.js#L36) flag of a dialog to fix this.